tests/qasm3/test_preprocess.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2025 qBraid
```

Comment:
Let's add a preprocessing test for qasm2 programs as well. It doesn't have to be an elaborate test set as we are already testing all of the processing cases in the qasm3 tests. This is just to ensure backward compatibility with qasm2 programs which might include `qelib1.inc` files.